src/renderers/ux-dialog-renderer.ts

```diff
@@ -183,13 +183,15 @@ export class DialogRenderer implements Renderer {
         dialogController.cancel();
       }
     };
-    this.dialogContainer.addEventListener('click', this.closeDialogClick);
-    this.anchor.addEventListener('click', this.stopPropagation);
+    const mouseEventType: MouseEventType = dialogController.settings.mouseEventType || 'click';
+    this.dialogContainer.addEventListener(mouseEventType, this.closeDialogClick);
+    this.anchor.addEventListener(mouseEventType, this.stopPropagation);
   }
 
-  private clearClickHandling(): void {
-    this.dialogContainer.removeEventListener('click', this.closeDialogClick);
-    this.anchor.removeEventListener('click', this.stopPropagation);
+  private clearClickHandling(dialogController: DialogController): void {
```

Comment:
I would add the event type as parameter, instead of the complete dialogController. It only needs the click event!

Reply:
Agree with this comment, method name can be changed from `clearClickHandling` to `clearMouseHandling`, or `clearEventHandling` for symmetry with the add. Or we can change both of them

Reply:
In the native-dialog-renderer the whole dialogController is also passed as an argument.

The functions there are xEventHandling, so your last comment will be reworked in the next patch, but the first one not.